dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserGroupService.java

```diff
@@ -216,7 +217,7 @@ public void updateUserGroups( User user, @Nonnull Collection<String> uids, User
             if ( canAddOrRemoveMember( userGroup.getUid(), currentUser ) )
             {
                 userGroup.addUser( user );
-                userGroupStore.updateNoAcl( userGroup );
```

Comment:
I seems wrong that the no ACL variant would behave differently other than not making the ACL checks. If that is the case maybe we should talk about changing that and making it behave the same.